experimental/dds/tree2/src/feature-libraries/modular-schema/modularChangeCodecs.ts

```diff
@@ -177,16 +180,63 @@ function makeV0Codec(
 			return undefined;
 		}
 		const list: [RevisionTag | undefined, ChangesetLocalId, any][] = encoded.map((tuple) =>
-			tuple.length === 3 ? tuple : [undefined, ...tuple],
+			tuple.length === 3
+				? [revisionTagCodec.decode(tuple[0]), tuple[1], tuple[2]]
+				: [undefined, ...tuple],
 		);
 		return nestedMapFromFlatList(list);
 	}
 
+	function encodeRevisionInfos(revisions: RevisionInfo[]): RevisionInfo[] {
+		const encodedRevisions = [];
+		for (const revision of revisions) {
+			if (revision.rollbackOf !== undefined) {
+				encodedRevisions.push({
+					...revision,
```

Comment:
When it comes to encoding and decoding I think we are supposed to avoid doing things like `...revision`. Otherwise one can end up changing what gets encoded by making a change to the in-memory format. There are no other members to this object anyway, right?

Reply:
+1--the docs I have a [PR out for](https://github.com/microsoft/FluidFramework/pull/18528) call out specifically do not do this during *encoding* (decoding is ok if it's convenient; less of a risk there)